api/finished_books.ts

```diff
@@ -0,0 +1,28 @@
+import { IListItem } from "../Interfaces/interfaces";
+
+export interface IFinishedBooksProps {
+  setIsLoading: (isLoading: boolean) => void;
+  setFinishedBooks: Array<IListItem>;
+}
+
+export const fetchFinishedBooks = ({
+  setIsLoading,
+  setFinishedBooks,
+}: IFinishedBooksProps) => {

```

Comment:
```suggestion
}: IFinishedBooksProps): Promise<IListItem[]> => {

```

Can you try this return function type, so that it'll be easier to map out the contents of a successful promise.